File: app/models/ranklist.rb
class Ranklist < ApplicationRecord
    belongs_to :user

    validates :program_name, presence: true
    validates :geographic, length: {in: 1..5, message: "Rating scale is between 1 through 5"}
    validates :goodfit, length: {in: 1..5, message: "Rating scale is between 1 through 5"}
    validates :reputation, length: {in: 1..5, message: "Rating scale is between 1 through 5"}
    validates :personalities, length: {in: 1..5, message: "Rating scale is between 1 through 5"}
    validates :edu_training, length: {in: 1..5, message: "Rating scale is between 1 through 5"}
    validates :user_id, length: {in: 1..5, message: "Rating scale is between 1 through 5"}

    attr_accessor :geographic, :goodfit, :reputation, :personalities, :edu_training

    def overall geographic, goodfit, reputation, personalities, edu_training
        geographic + goodfit + reputation + personalities + edu_training
    end

end

File: client/src/components/User/Ranklist/Ranklist.js
import React, { useState, useContext } from 'react'
import { useParams } from 'react-router-dom';
import { UserContext } from '../../../context/userContext'
import NewRankCard from './NewRankCard';
import RanklistCard from './RanklistCard';

function Ranklist() {
    const { users } = useContext(UserContext)
    const [isShow, setIsShow] = useState(false);

    const getRanklists = users.ranklists.map((ranklist) => {
        return <div className="ranklist-preview" key={ranklist.id}>
            <RanklistCard
                id={ranklist.id}
                program_name={ranklist.program_name}
                geographic={ranklist.geographic}
                goodfit={ranklist.goodfit}
                reputation={ranklist.reputation}
                personalities={ranklist.personalities}
                qol={ranklist.qol}
                edu_training={ranklist.edu_training}
                comment={ranklist.comment}
                user_id={ranklist.user_id}
                ranklist={ranklist}
            />
        </div>
    })

    return (
        <div>
            <h1>{users.first_name}'s rank list</h1>
            {isShow ? (
                <>
                    <NewRankCard setIsShow={setIsShow} isShow={isShow} />
                    <button onClick={() => setIsShow((isShow) => !isShow)}>
                        <span role="img" aria-label="edit">
                            ❌ Cancel
                        </span>
                    </button>
                </>
            ) : (

                <button onClick={() => setIsShow((isShow) => !isShow)}>
                    <span role="img" aria-label="edit">
                        📝Add Program
                    </span>
                </button>

            )}
            <ul>
                {getRanklists}
            </ul>
        </div>
    )
}

export default Ranklist
